WikiWatershed/model-my-watershed
# Monitor: Add BiG-CZ Search

## Overview

Adds the BiG-CZ search functionality to the Monitor tab, and styles it according to the mockups. Harmonizes behavior with rest of MMW, so that switching between Analyze and Monitor, going from Analyze / Monitor to Draw, and to Model, behaves as expected.

The Model stage currently does not have the Monitor tab. That will be added in #2621.

These upgrades break BiG-CZ. That will be fixed in #2639.

Connects #2622 

### Demo

* Overview

    ![2018-02-15 09 41 58](https://user-images.githubusercontent.com/1430060/36262607-f6d442e8-1235-11e8-865d-883b799e120e.gif)

* Analyze Tab

    ![screen shot 2018-02-15 at 9 45 04 am](https://user-images.githubusercontent.com/1430060/36263448-458bbae0-1238-11e8-8528-f8d940ead569.png)

* Monitor Tab (with search results)

    ![screen shot 2018-02-15 at 9 45 09 am](https://user-images.githubusercontent.com/1430060/36263471-4f7a057a-1238-11e8-86b2-8d0129fbeeff.png)

* Monitor Tab (with filter sidebar)

    ![screen shot 2018-02-15 at 9 45 14 am](https://user-images.githubusercontent.com/1430060/36263499-58cd3782-1238-11e8-96fa-c3df66c2c030.png)

* Monitor Tab (with detail view)

    ![screen shot 2018-02-15 at 9 45 24 am](https://user-images.githubusercontent.com/1430060/36263570-7966eba0-1238-11e8-9085-9c7a4dae9d7f.png)
    ![screen shot 2018-02-15 at 9 46 02 am](https://user-images.githubusercontent.com/1430060/36263575-7c261de8-1238-11e8-8187-645c7be768e2.png)

Tagging @ajrobbins and @jfrankl for visual review.

### Notes

The screen capture software I use slows down the browser. In reality the interactions are faster than the ones seen above.

There were no mockups for the filter sidebar, so I made it fit as best as I could, but we may need to create a follow up card for design tweaks there.

Should there be a spinner under the Analyze and Monitor tab headers when they are loading?

## Testing Instructions

* Check out this branch and `bundle`
* Go to [:8000/](http://localhost:8000/) and select / draw a shape
* See the Monitor tab in the Analyze stage. Click it and search for something. Ensure the search works as expected.
* Test the filters. Ensure they work as expected.
* Ensure items on the map are in sync with those in the sidebar.
* Click an item in the sidebar. Ensure the details view is shown, and the map only shows that items shape, and you can scroll the details view if it is too long.
* Ensure pagination works as expected.
* Ensure switching to Analyze hides the Monitor results from the map.
* Ensure switching back to Monitor restores the results on the map.
* Ensure tab state (which analyses is selected, which catalog is selected, which result is selected, scroll state) remains consistent when switching between Analyze and Monitor.
* Ensure that any Monitor specific elements (map popup when multiple results, filter sidebar) are removed when switching to Analyze.
* Ensure that clicking "Change Area" behaves as expected, and when revisiting Monitor you get a blank slate for a new search.
* Ensure that selecting a model and moving on to Model removes the Monitor elements from the map.
* Ensure that when selecting "New Project" from the Model and going back to Monitor, you get a blank slate for a new search.
* Ensure that selecting "New Project" from the Projects list and going back to Monitor, you get a blank slate for a new search.
* Ensure the visual look and feel matches and feels harmonious, and follows [the mockups here](https://app.goabstract.com/projects/76ed1390-dea8-11e7-a27c-0f868efbc56f/branches/714a77e0-fa21-11e7-835a-4527db42429f/files/7F4A25A6-1E0D-48A0-B999-8629B42E4E23/pages/6F1410C0-7988-41C6-849F-1AAB98853CD7?changed=true)
